lib/Dialect/FIRRTL/Import/FIRParser.cpp

```diff
@@ -2591,22 +2596,22 @@ ParseResult FIRStmtParser::parseWhen(unsigned whenIndent) {
 
     // We parse the substatements into their own parser, so they get inserted
     // into the specified 'when' region.
-    FIRStmtParser subParser(blockToInsertInto, moduleContext, modNameSpace,
-                            version);
+    auto *subParser = new (moduleContext.subParserAllocator)
```

Comment:
> LazyLocationListener locationProcessor)

Ah, nice catch. Thanks!

Reply:
> Alternatively, store them in a deque or something (stable storage, runs destructors, add with emplace_back if needed, etc.), removing them from the back on exit (scoped_exit? shrug) to keep down peak memory usage-- for example, if the deeply nested structures occur multiple times in a module it'd be nice to not continue to grow storage for these. That said these are fairly cheap so up to you!

Tried this but it requires decent changes (mostly NFC though). FIRStmtParser has a reference to moduleContext so we have to reorder FIRStmtParser and ModuleContext in order to declare stack of FIRStmtParsesr in ModuleContext but I feel it's not much worth making the implementation complicated just because of the micro optimization of subparser. I'll just use unique_ptr and revisit if that causes performance regression.